SmartEVSE-3/src/evse.cpp

```diff
@@ -1011,128 +1035,193 @@ void CalcBalancedCurrent(char mod) {
         Baseload = 0;
+    }
+
+    // Room for extra current (per phase) upto any of the max settings: MaxMains, MaxCircuit en MaxSumMains
+    // May be negative in case not enough power is available
+    int16_t MaxMainsRoom = (MaxMains * 10) - Imeasured;
+    int16_t MaxCircuitRoom = (MaxCircuit * 10) - Imeasured_EV;
+    int16_t MaxSumMainsRoom = ((MaxSumMains * 10) - Isum)/EstimateNrOfPhasesCharging(); //convert from sum of phases to current per phase
+    Idifference = min(MaxMainsRoom, min(MaxCircuitRoom, MaxSumMainsRoom));
+    _LOG_V("Checkpoint 2a Idifference=%.1f is minimum of MaxMainsRoom=%.1f MaxCircuitRoom=%.1f MaxSumMainsRoom=%.1f.\n", (float)Idifference/10, (float)MaxMainsRoom/10, (float)MaxCircuitRoom/10, (float)MaxSumMainsRoom/10);
+    // Maximum current (per phase) from baseload upto  any of the max settings: MaxMains, MaxCircuit en MaxSumMains
+
+    int16_t MaxMainsAvailable = (MaxMains * 10) - Baseload;
+    int16_t MaxCircuitAvailable = (MaxCircuit * 10) - Baseload_EV;
+    int16_t MaxSumMainsAvailable = ((MaxSumMains * 10)/EstimateNrOfPhasesCharging()) - Baseload;
+    ImaxAvailable = min(MaxMainsAvailable, min(MaxCircuitAvailable, MaxSumMainsAvailable));
+    _LOG_V("Checkpoint 2b ImaxAvailable=%.1f is minimum of MaxMainsAvailable=%.1f MaxCircuitAvailable=%.1f MaxSumMainsAvailable=%.1f.\n", (float)ImaxAvailable/10, (float)MaxMainsAvailable/10, (float)MaxCircuitAvailable/10, (float)MaxSumMainsAvailable/10);
 
     if (Mode == MODE_NORMAL)                                                    // Normal Mode
     {
-        if (LoadBl == 1)                                                        // Load Balancing = Master? MaxCircuit is max current for all active EVSE's;
-            IsetBalanced = MaxCircuit * 10 - Baseload_EV;                       // subpanel option not valid in Normal Mode;
-                                                                                // limiting is per phase so no Nr_Of_Phases_Charging here!
-        else
-            IsetBalanced = ChargeCurrent;                                       // No Load Balancing in Normal Mode. Set current to ChargeCurrent (fix: v2.05)
-        if (ActiveEVSE && mod) {                                                // Only if we have active EVSE's and New EVSE charging
-            // Set max combined charge current to MaxMains - Baseload, or MaxCircuit - Baseload_EV if that is less
-            IsetBalanced = min((MaxMains * 10) - Baseload, (MaxCircuit * 10 ) - Baseload_EV); //TODO: why are we checking MaxMains and MaxCircuit while we are in Normal mode?
-                                                                                              //TODO: capacity rate limiting here?
+        IsetBalanced = ChargeCurrent;           // Set current to ChargeCurrent (fix: v2.05)
+        if (LoadBl == 1 || (ActiveEVSE && mod)) // Load Balancing = Master or new EVSE and already active;
+        {
+            IsetBalanced = ImaxAvailable;
```

Comment:
This is incorrect; in Normal Mode the settings MaxMains is not even available in configuration, so it should not be taken into account; users who only use Normal Mode would be confronted with the default settings of MaxMains, which would lead to questions/problems....

Reply:
thanks, I will review